packages/lambdas/src/sqs-to-converter.ts

```diff
@@ -511,27 +510,49 @@ async function storePreProcessedConversionResult({
   }
 }
 
-function buildDocumentNameForPartialConversions(
-  fileName: string,
-  numChunksInPayload: number,
-  index: number
-): string {
-  if (numChunksInPayload <= 1) return fileName;
+function buildDocumentNameForPartialConversions(fileName: string, index: number): string {
+  const paddedIndex = index.toString().padStart(2, "0");
+  return `${fileName}_part_${paddedIndex}.xml`;
+}
 
-  return `${fileName}_part_${index}.xml`;
+async function storePartitionedPayloadsInS3({
+  partitionedPayload,
+  preConversionFilename,
+  message,
+  lambdaParams,
+  log,
+}: {
+  partitionedPayload: string[];
+  preConversionFilename: string;
+  message: SQSRecord;
+  lambdaParams: Record<string, string | undefined>;
+  log: typeof console.log;
+}) {
+  partitionedPayload.forEach((payload, index) => {
+    storePayloadInS3({
+      payload,
+      fileName: buildDocumentNameForPartialConversions(preConversionFilename, index),
+      message,
+      lambdaParams,
+      type: "pre-convert",
+      log,
+    });
+  });
 }
 
-async function storePreConversionPayloadInS3({
+async function storePayloadInS3({
   payload,
-  preConversionFilename: preProcessedFilename,
+  fileName,
   message,
   lambdaParams,
+  type,
   log,
 }: {
   payload: string;
-  preConversionFilename: string;
+  fileName: string;
   message: SQSRecord;
   lambdaParams: Record<string, string | undefined>;
+  type: "pre-convert" | "clean";
```

Comment:
`type` is only being used for alerting/debugging, right? But doesn't the filename indicate the type of file already?

Reply:
what's the effect tho? do you think i should check the file name to detect the type?

Reply:
No, I mean the code doesn't do anything with the type, its only there for our debugging, right? In which case, the name already indicates the type of file.

But if we're using the type for something then its fine to keep it around - I think we're not, but cool if we are.

Reply:
Ah ya. I just wanted to keep it so it separates the msg for sentry, but on the second thought, maybe it's better not to, since this is the same type of issue anyway, and like you said, the file name already tells us where the issue happened. So good catch, and ill remove it